### ISSUE TITLE
[Bug] ShadowPEFT sequence classification pools the wrong token with left padding

### ISSUE THREAD

**Issue by aswanth-07** [huggingface/peft#3620]
### System Info

- PEFT `main`: `9c16ee66cd4c58bd9cdf2d8b4e06c1cf8e8f8efe`
- Affected code: the ShadowPEFT sequence-classification pooling path introduced in #3354.

### Who can help?

@BenjaminBossan

### Reproduction

ShadowPEFT documents `_pool_last_token` as selecting the last non-padding representation, but it derives the index as `attention_mask.sum(dim=1) - 1`:

```python
import torch

from peft.tuners.shadow.model import _pool_last_token

hidden = torch.tensor(
    [
        [[10.0], [20.0], [30.0], [40.0]],
        [[10.0], [20.0], [30.0], [40.0]],
    ]
)
attention_mask = torch.tensor(
    [
        [0, 0, 1, 1],
        [0, 1, 1, 1],
    ]
)

print(_pool_last_token(hidden, attention_mask).squeeze(-1))
```

The current index expression selects positions 1 and 2, producing `[20, 30]`. Both are incorrect for these left-padded rows. The last non-padding position is 3 in each row, so the expected result is `[40, 40]`.

The same count-based implementation appears twice:

- the helper used by `ShadowModel.shadow_auxiliary_loss`, so the auxiliary sequence-classification loss trains against a padding position;
- `DetachedShadowModel._pool_last_token`, so `unload_shadow()` inference returns logits from the wrong position.

Transformers sequence-classification models intentionally select the rightmost non-padding position to support both left and right padding:

https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_layers.py

Tokenizer configuration also exposes `padding_side` as a supported choice:

https://huggingface.co/docs/transformers/main_classes/tokenizer

I searched PEFT issues, pull requests, and history for ShadowPEFT padding, classification pooling, left padding, and `_pool_last_token`. I found no overlapping report or implementation.

### Expected behavior

ShadowPEFT sequence classification should select the rightmost position whose attention-mask value is nonzero, matching Transformers for both left- and right-padded batches.

The proposed scope is one shared pooling helper used by both training and detached inference, with focused regressions for left and right padding. Causal-language-model and cache behavior would remain unchanged.

Would you like me to prepare that focused PR? I will wait for explicit approval before changing code.

AI assistance was used during the source audit and duplicate search. I reviewed the complete affected path and independently checked the index calculation against the documented Transformers behavior.

**Comment by Cunninger** [comment]
I'll take this. `_pool_last_token` indexes with `attention_mask.sum(dim=1) - 1`, which is wrong under left padding. I'll select the last non-padding token and add a regression test for left-padded batches.

**Comment by Cunninger** [comment]
PR: https://github.com/huggingface/peft/pull/3622

**Comment by BenjaminBossan** [comment]
Thanks for reporting this issue @aswanth-07 and for awaiting approval first. Please go ahead and create the PR.

**Comment by aswanth-07** [comment]
Thanks for the approval! I opened #3629 with a shared pooling helper and regression coverage for both the auxiliary loss and detached classifier paths. :)
